compair/static/modules/rich-content/rich-content-directive.js

```diff
@@ -318,7 +318,9 @@ module.directive('richContent',
 
                     // needs to be compiled by dynamic
                     $scope.richContent = '<div mathjax hljs >'+content+'</div>';
+                };
 
+                var processAttachment = function() {
                     $scope.attachmentContent = [];
                     if ($scope.attachment) {
                         $scope.attachmentContent.push(
```

Comment:
Does this also cover removing attachments? I'm not entirely sure how this works, but just asking this since you're doing a push, rather than an update.

Reply:
setting `$scope.attachmentContent = [];` will remove all attachments from scope every time `processAttachment` is called. The if condition will then add the attachment data if `$scope.attachment` has a value